doc/BGP/BGP-supress-fib-pending.md

```diff
@@ -0,0 +1,799 @@
+```jinja2
+router bgp {{ DEVICE_METADATA['localhost']['bgp_asn'] }}
+  bgp suppress-fib-pending
+```
+
+Zebra shall be started with an option ```--asic-offload=notify_on_offload```. This option will make zebra wait for ```RTM_F_OFFLOAD``` before notifying about installation status to bgpd. The template *supervisor.conf.j2* file is updated:
+
+```jinja2
+[program:zebra]
+command=/usr/lib/frr/zebra -A 127.0.0.1 -s 90000000 -M dplane_fpm_nl --asic-offload=notify_on_offload
+```
+
+Before the route is offloaded ```show ip route``` shows routes as queued:
+
+```
+admin@sonic:~$ show ip route
+Codes: K - kernel route, C - connected, S - static, R - RIP,
+       O - OSPF, I - IS-IS, B - BGP, E - EIGRP, N - NHRP,
+       T - Table, v - VNC, V - VNC-Direct, A - Babel, F - PBR,
+       f - OpenFabric,
+       > - selected route, * - FIB route, q - queued, r - rejected, b - backup
+       t - trapped, o - offload failure
+
+B>q 0.0.0.0/0 [20/0] via 10.0.0.1, PortChannel102, weight 1, 00:04:46
+  q                  via 10.0.0.5, PortChannel105, weight 1, 00:04:46
+  q                  via 10.0.0.9, PortChannel108, weight 1, 00:04:46
+  q                  via 10.0.0.13, PortChannel1011, weight 1, 00:04:46
+```
+
+Once zebra is notified about successfull route offload status the *offload* flag is set:
+
+```
+admin@sonic:~$ vtysh -c "show ip route 100.1.0.25/32 json"
+{
+    "100.1.0.25/32": [
+        {
+            "prefix": "100.1.0.25/32",
+            "prefixLen": 32,
+            "protocol": "bgp",
+            "vrfId": 0,
+            "vrfName": "default",
+            "selected": true,
+            "destSelected": true,
+            "distance": 20,
+            "metric": 0,
+            "installed": true,
+            "offloaded": true,
+            "table": 254,
+            "internalStatus": 80,
+            "internalFlags": 264,
+            "internalNextHopNum": 1,
+            "internalNextHopActiveNum": 1,
+            "nexthopGroupId": 4890,
+            "installedNexthopGroupId": 4890,
+            "uptime": "00:22:20",
+            "nexthops": [
+                {
+                    "flags": 3,
+                    "fib": true,
+                    "ip": "10.0.0.49",
+                    "afi": "ipv4",
+                    "interfaceIndex": 788,
+                    "interfaceName": "PortChannel1013",
+                    "active": true,
+                    "weight": 1
+                }
+            ]
+        }
+    ]
+}
+```
+
+#### 7.3. RouteOrch
+
+<!-- omit in toc -->
+#### ResponsePublisher in RouteOrch
+
+*RouteOrch* need to use existing *ResponsePublisher* API to be able to publish responses into *APPL_STATE_DB*.
+
+A snippet of *ResponsePublisher* API that is going to be used is given below.
+
+```c++
+void ResponsePublisher::publish(const std::string &table, const std::string &key,
+                                const std::vector<swss::FieldValueTuple> &intent_attrs,
+                                const ReturnCode &status,

```

Comment:
>status [](http://example.com/codeflow?start=50&length=6)

In the API parameter, there is a `ReturnCode`. How is it used? Why it is missing in below Redis example?

Reply:
This API is not new, it already exists in SONiC (https://github.com/sonic-net/sonic-swss/blob/master/orchagent/response_publisher.h#L38).

The return code is sent as part of notification https://github.com/sonic-net/sonic-swss/blob/master/orchagent/response_publisher.cpp#L117 but not recorded in APPL STATE DB for some reason.

For proposed functionality this is not required anyway.